### PR TITLE
binlog, relay: support binlog sequence number larger than 999999

### LINKED ISSUE #1475: use numeric comparison when compare binlog filename
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

make upstream MySQL generate `mysql-bin.1000000`, some logic may not well handled because they use string comparison. For example, https://github.com/siddontang/go-mysql/blob/0c5789dd0bd378b4b84f99b320a2d35a80d8858f/mysql/position.go#L13, https://github.com/pingcap/dm/blob/511cea1ea51822274f6ad7fe4acdc8ec948de8f3/relay/writer/file.go#L242-L244

2. What did you expect to see?
use numeric comparison (and go-mysql is not updated for a long time, maybe we could use our forked version)

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

close #1475 

### What is changed and how it works?
compare by numeric logic, not byte order

### Check List 

Tests 

 - integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

**Comment by lance6716** [comment]
I'll find a way to do an integration test 😵

**Comment by lance6716** [comment]
need adjust `|<upstream switch number>` in another PR

**Comment by lance6716** [comment]
we could inspect docker mount path and modify binlog files. will add integration test later

https://dev.mysql.com/doc/refman/5.7/en/docker-mysql-more-topics.html

**Comment by csuzhangxc** [comment]
> we could inspect docker mount path and modify binlog files. will add integration test later
> 
> https://dev.mysql.com/doc/refman/5.7/en/docker-mysql-more-topics.html

oh, you can even build a special MySQL docker image with pre-generated binlog files.

File: .github/workflows/binlog-999999.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          sudo cp -r -L /tmp/dm_test/incremental_mode/worker2/log ./logs/worker2
+          sudo chown -R runner ./logs
+
+      # Update logs as artifact seems not stable, so we set `continue-on-error: true` here.
+      - name: Upload logs
+        continue-on-error: true
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: upstream-switch-logs
```
Comment (GMHDBJD) [inline .github/workflows/binlog-999999.yml:93]: change name

File: go.mod [hunk 2]
```diff
@@ -49,3 +49,5 @@ require (
 )
 
 go 1.13
+
+replace github.com/siddontang/go-mysql v1.1.1-0.20200824131207-0c5789dd0bd3 => github.com/lance6716/go-mysql v1.1.1-0.20210303100354-b0e44c2c5623
```
Comment (GMHDBJD) [inline go.mod:53]: Will the pr merged to go-mysql?

File: tests/binlog_999999/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  mysql2: # upstream MySQL-2
+    <<: *default-node
+    container_name: mysql2
+    hostname: mysql2
+    ports:
+      - "3307:3306"
+    networks:
+      db-networks:
+        ipv4_address: 172.28.128.5
+    image: mysql:5.7.32
```
Comment (GMHDBJD) [inline tests/binlog_999999/docker-compose.yml:40]: Maybe change to 8.0?

**Review by GMHDBJD — COMMENTED** [review]
rest LGTM

File: go.mod (hunk 2 above)
Comment (lance6716) [reply go.mod:53]: https://github.com/siddontang/go-mysql is not maintained now. maybe we could give a hand after finishing our works 😂

**Comment by lance6716** [comment]
ping @GMHDBJD

**Review by GMHDBJD — APPROVED** [review]
/lgtm

**Pull request merged**